rust/rust.bzl

```diff
@@ -172,8 +186,7 @@ def _setup_deps(deps, name, working_dir, allow_cc_deps=False,
       libs += native_libs
       transitive_libs += native_libs
       symlinked_libs += native_libs
-      link_flags += ["-l static=" + dep.label.name]
-      has_native = True
+      native_link_flags += ["-l static={}".format(_native_lib_name(native_lib)) for native_lib in dep.cc.libs]
```

Comment:
How does this fail on dylibs? Looks like maybe a linker error on build?

Reply:
Probably a good idea. I'm not super familiar with what kinds of things you can put in a depset, but if I can put generic strings in a depset than I think that's the right way to do this instead of appending to a list.

I think when I tried using this in a different codebase (which has dylibs in the dep chain) it failed with undefined references, but it was actually caused by the `_native_lib_name()` function being too primitive. It doesn't take into account dylibs with SONAME extensions, like `libfoo.so.3.0`. Plus it really needs to link those with `-l dynamic=` instead of `-l static=`.

I'm not quite sure how to pass those dylibs to rustc. When Bazel encounters an SONAME dylib it links with `-l:libfoo.so.3.0` instead of `-lfoo` to specify the exact .so it wants to link with. Maybe for those dylibs we should pass through the `-l:` form as a raw linker arg instead of using rustc's library linking flags.